lib/src/license_detection/new_tokenizer.dart

```diff
@@ -0,0 +1,278 @@
+import 'dart:collection';
+
+import 'package:meta/meta.dart';
+import 'package:source_span/source_span.dart';
+import 'package:string_scanner/string_scanner.dart';
+
+@sealed
+class NewToken {
+  final String normalizedText;
+  final int tokenID;
+  final int line;
+  final FileSpan fileSpan;
```

Comment:
I practice it'll be a `FileSpan`, but `FileSpan` is a subclass of `SourceSpan` and isn't it better to use the generic interface if we only need that?
```suggestion
  final SourceSpan span;
```